### ISSUE TITLE
oidc-gen error on C7 

### ISSUE THREAD

**Issue by andrea-manzi** [indigo-dc/oidc-agent#421]
Hello,
i have installed the latest version of oidc-agent for C7 from the repo  https://repo.data.kit.edu/data-kit-edu-centos7.repo 

but i have an issue when running this command:

`oidc-gen --pub --issuer https://aai.egi.eu/oidc --scope "openid profile offline_access eduperson_entitlement" andrea`

the qrcode displays but then i see

`Error: Argument is NULL in function getAccountFromJSON`

is it a known problem?

thanks!
Andrea

**Comment by andrea-manzi** [comment]
sorry the oidc-agent was not running ( i started it but then i updated the package to a new version and it was not restarted) sorry for the noise!

**Comment by marcvs** [comment]
Thanks for reporting. I guess we could check if the agent is running and provide a more helpful error message.

**Comment by zachmann** [comment]
I'm a bit surprised by this bug - or the fact that this is due to the agent not running. I would assume that something else was going on, which we cannot reproduce anymore.

Generally, the `oidc-add`, `oidc-token`, and `oidc-gen` tools check if the agent is running before doing operations that involve the agent and give an error if they cannot connect to the agent. So if the agent wasn't running when `oidc-gen` was called there would have been an error message indicating that.

I tried your command and terminated the agent while the authentication was pending:
- For the auth code flow `oidc-gen` terminated with an error message `Could not connect to oidc-agent`, so everything fine there.
- For the device code flow `oidc-gen` terminated with a more cryptic error message:
```
Could not decode json: (null)
This seems to be a bug. Please hand in a bug report.
Error: Argument is NULL in function getJSONValuesFromString
```

While this looks similar to your error, I assume that the two first lines are additional and weren't present with your error and the function within which the argument is NULL is different.

I cannot reproduce your exact error and also don't have an idea where it might come from, but I'll improve the error I found.
